src/rpm.rs

```diff
@@ -382,28 +456,54 @@ where
 }
 
 impl Header<IndexSignatureTag> {
-    fn new_signature_header(size: i32, md5: &[u8], sha1: String) -> Self {
+    /// creates a new full signature header
+    ///
+    /// `size` is combined size of header, header store and the payload
+    ///
+    /// PGP and RSA tags expect signatures according to [RFC2440](https://tools.ietf.org/html/rfc2440)
+    fn new_signature_header(
+        size: i32,
+        md5sum: &[u8],
+        sha1: String,
+        rsa_spanning_header: &[u8],
+        rsa_spanning_header_and_archive: &[u8],
+    ) -> Self {
         let offset = 0;
```

Comment:
this implementation seems redundant. Can you use the signature_builder here?

Reply:
Good catch!